Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -207,6 +235,86 @@ int MatchTOF::prepareInteractionTimes()
   // do nothing. If you think it can be useful have a look at MatchTPCITS
   return 0;
 }
+//______________________________________________
+void MatchTOF::printGrouping(const std::vector<o2::dataformats::MatchInfoTOFReco>& origin, const std::vector<std::vector<o2::dataformats::MatchInfoTOFReco>>& grouped)
+{
+  printf("Original vector\n");
+  for (const auto& seed : origin) {
+    printf("Pair: track=%d TOFcl=%d\n", seed.getIdLocal(), seed.getTOFClIndex());
+  }
+
+  printf("\nGroups\n");
+  int ngroups = 0;
+  for (const auto& gr : grouped) {
+    ngroups++;
+    printf("Group %d\n", ngroups);
+    for (const auto& seed : gr) {
+      printf("Pair: track=%d TOFcl=%d\n", seed.getIdLocal(), seed.getTOFClIndex());
+    }
+  }
+}
+//______________________________________________
+void MatchTOF::groupingMatch(std::vector<o2::dataformats::MatchInfoTOFReco> origin, std::vector<std::vector<o2::dataformats::MatchInfoTOFReco>>& grouped, std::vector<std::vector<int>>& firstEls, std::vector<std::vector<int>>& secondEls)
+{
+  grouped.clear();
+  firstEls.clear();
+  secondEls.clear();
+
+  std::vector<o2::dataformats::MatchInfoTOFReco> dummy;
+  std::vector<int> dummyInt;
+
+  int pos = 0;
+  while (origin.size()) { // go ahead if there are elements
+    bool found = true;
+    grouped.push_back(dummy);
+    auto& trkId = firstEls.emplace_back(dummyInt);
+    auto& cluId = secondEls.emplace_back(dummyInt);
+
+    // first element is the seed
+    const auto& seed = origin[0];
+    trkId.push_back(seed.getIdLocal());
+    cluId.push_back(seed.getTOFClIndex());
+    // remove element added to the current group
+    grouped[pos].push_back(seed);
+    origin.erase(origin.begin());
```

Comment:
`vector.erase` is very expensive method, can't you avoid it, e.g. by using some flags? If not, then, since you are always erasing the 1st element, it is better to use std::dequeu, which has pop_front method.

Reply:
using flags (unfortunately erase was called also in an internal loop to delete object in random positions)